avm/utilities/pipelines/staticValidation/compliance/module.tests.ps1

```diff
@@ -954,6 +907,184 @@ Describe 'Module tests' -Tag 'Module' {
+        templateFileContentBicep = Get-Content $templateFilePath
+      }
+
+      # Setting expected URL only for those that doen't have multiple different variants
+      $avmInterfaceSpecsTemplateBase = 'https://raw.githubusercontent.com/Azure/Azure-Verified-Modules/main/docs/static/includes/interfaces'
+      $avmInterfaceSpecsBase = 'https://azure.github.io/Azure-Verified-Modules/specs/shared/interfaces'
+      $udtCases = @(
+        @{
+          parameterName = 'diagnosticSettings'
+          udtName       = 'diagnosticSettingType'
+          link          = "$avmInterfaceSpecsBase#diagnostic-settings"
+        }
+        @{
+          parameterName  = 'roleAssignments'
+          udtName        = 'roleAssignmentType'
+          udtExpectedUrl = "$avmInterfaceSpecsTemplateBase/int.rbac.udt.schema.bicep"
+          link           = "$avmInterfaceSpecsBase#role-assignments"
+        }
+        @{
+          parameterName  = 'lock'
+          udtName        = 'lockType'
+          udtExpectedUrl = "$avmInterfaceSpecsTemplateBase/int.locks.udt.schema.bicep"
+          link           = "$avmInterfaceSpecsBase#resource-locks"
+        }
+        @{
+          parameterName = 'managedIdentities'
+          udtName       = 'managedIdentitiesType'
+          link          = "$avmInterfaceSpecsBase#managed-identities"
+        }
+        @{
+          parameterName = 'privateEndpoints'
+          udtName       = 'privateEndpointType'
+          link          = "$avmInterfaceSpecsBase#private-endpoints"
+        }
+        @{
+          parameterName = 'customerManagedKey'
+          udtName       = 'customerManagedKeyType'
+          link          = "$avmInterfaceSpecsBase#customer-managed-keys"
+        }
+      )
+
+      foreach ($udtCase in $udtCases) {
+        $udtTestCases += @{
+          moduleFolderName         = $resourceTypeIdentifier
+          templateFileContent      = $templateContent
+          templateFileContentBicep = Get-Content $templateFilePath
+          parameterName            = $udtCase.parameterName
+          udtName                  = $udtCase.udtName
+          expectedUdtUrl           = $udtCase.udtExpectedUrl ? $udtCase.udtExpectedUrl : ''
+          link                     = $udtCase.link
+        }
+      }
+    }
+
+
+    It "[<moduleFolderName>] If template has a parameter [<parameterName>], it should implement the user-defined type [<udtName>]" -TestCases $udtTestCases {
+
+      param(
+        [hashtable] $templateFileContent,
+        [string[]] $templateFileContentBicep,
+        [string] $parameterName,
+        [string] $udtName,
+        [string] $expectedUdtUrl,
+        [string] $link
+      )
+
+      if ($templateFileContent.parameters.Keys -contains $parameterName) {
+        $templateFileContent.parameters.$parameterName.Keys | Should -Contain '$ref' -Because "the [$parameterName] parameter should use a user-defined type. For for information please review the [AVM Specs]($link)."
```

Comment:
```suggestion
        $templateFileContent.parameters.$parameterName.Keys | Should -Contain '$ref' -Because "the [$parameterName] parameter should use a user-defined type. For information please review the [AVM Specs]($link)."
```